calour/io.py

```diff
@@ -334,6 +347,14 @@ def read(data_file, sample_metadata_file=None, feature_metadata_file=None,
     exp = cls(data, sample_metadata, feature_metadata,
               exp_metadata=exp_metadata, description=description, sparse=sparse)
 
+    # remove samples containing NaN in data table (can happen in _read_csv)
+    nan_samples = np.where(np.isnan(exp.data.sum(axis=1)))[0]
```

Comment:
similarly, we wanna minimize the changes in the read(). we shouldn't remove any if it's in the data file. For example, what if one wants to fill na with zero or impute with mean.

Reply:
if it is needed, then it should be a function in filtering module to drop na.

Reply:
added in filtering.py
however, we need to call it before the normalization in read(), so i don't know how to enable users to read data with nans (since normalize fails with nans).
The problem is mzmine2 data contains a comma at the end of each line, which leads to a sample with nans in all features when reading using pandas.read_csv()
therefore, we need to remove them in order for normalization in the read() to work...
options are:
1. always filter the nans (current implementation - now via calling the filter_data_na() from filtering.py in read()). problem is it will do it for all data, so will remove nans from all tables.
2. add a flag to say whether to filter nans or not for the read() function, and then filter/not filter according to it, and set it to true when reading mzmine2 data
3. fix the _read_csv() function to specifically get rid if there is a last empty column
what do you think is best? or any other idea?